server/portal/apps/tickets/utils.py

```diff
@@ -0,0 +1,12 @@
+from django.conf import settings
+import requests
+def get_recaptcha_verification(request):
+    recaptcha_response = request.POST.get('recaptchaResponse')
+    secret_key = getattr(settings, 'RECAPTCHA_SECRET_KEY')
+    data = {
+        'secret': secret_key,
+        'response': recaptcha_response
+    }
+    r = requests.post('https://www.google.com/recaptcha/api/siteverify', data=data)
+    recap_result = r.json()
+    return recap_result
```

Comment:
Would there be situations where these requests have to be handled?

Reply:
I have to look at up how to use Recaptcha, but `requests` is basically an HTTP library that handle POST request for google Recaptcha API

Reply:
@williamkwon97 By handling, I meant checking the request (i.e. when the http response isn't a success code like 200 or whatever it responds on success).
Wouldn't there be a need to handle the error responses? For instance, if the google endpoint is down, it would not be handled on our side and only show a stack trace in the backend logs. Wouldn't it be better to handle it properly in case Google's stuff is down?

Reply:
@duckonomy hmm I see, I guess that case would be very rare but if when a request isn't valid then you get anything other than success, it will display 'Invalid reCAPTCHA. Please try again.'

Reply:
@williamkwon97 You could get a key error at https://github.com/TACC/Core-Portal/blob/91f8f953161b403ed4e3ce0516b7c822346b5887/server/portal/apps/tickets/api/views.py#L76 

Maybe change this:
```python
            recap_result = utils.get_recaptcha_verification(request)
            if recap_result['success']:
                ticket_id = rt.create_ticket(subject=subject,
                                         problem_description=problem_description,
                                         requestor=email,
                                         cc=cc,
                                         attachments=attachments)
                return JsonResponse({'ticket_id': ticket_id})
            if not recap_result['success']:
                raise ApiException('Invalid reCAPTCHA. Please try again.')
```
to this:
```python
            recap_result = utils.get_recaptcha_verification(request)
            if recap_result.get('success', False):
                ticket_id = rt.create_ticket(subject=subject,
                                         problem_description=problem_description,
                                         requestor=email,
                                         cc=cc,
                                         attachments=attachments)
                return JsonResponse({'ticket_id': ticket_id})
            else:
                raise ApiException('Invalid reCAPTCHA. Please try again.')
```

Reply:
@duckonomy that was my original code too but Sal recommends changing to current codes https://github.com/TACC/Core-Portal/pull/530#discussion_r745154153

Reply:
@williamkwon97 Gotcha. I was trying to get at this one though.
```python
            if recap_result.get('success', False):
```

Reply:
@duckonomy updated